pandas/tests/groupby/test_groupby.py

```diff
@@ -1714,3 +1714,36 @@ def test_groupby_multiindex_nat():
     result = ser.groupby(level=1).mean()
     expected = pd.Series([3., 2.5], index=["a", "b"])
     assert_series_equal(result, expected)
+
+
+def test_idxmin_idxmax_returns_int_types():
+    # GH 25444
+    df = pd.DataFrame({'name': ['A', 'A', 'B', 'B'],
+                       'c_int': [1, 2, 3, 4],
+                       'c_float': [4.02, 3.03, 2.04, 1.05],
+                       'c_date': ['2019', '2018', '2016', '2017']})
+    df['c_date'] = pd.to_datetime(df['c_date'])
+
+    idxmins_computed = df.groupby('name').idxmin()
+    idxmaxs_computed = df.groupby('name').idxmax()
+
+    for col in idxmaxs_computed:
+        assert idxmins_computed[col].dtype == np.int64
+        assert idxmaxs_computed[col].dtype == np.int64
+
+    idxmins_expected = pd.DataFrame({'c_int': [0, 2],
+                                     'c_float': [1, 3],
+                                     'c_date': [1, 2]},
+                                    index=['A', 'B'])
```

Comment:
put the name when you create the index, e.g. ``Index(['A', 'B'], name='name')``

Reply:
done in https://github.com/pandas-dev/pandas/pull/25531/commits/d874ef7adf0b5fdad7525ed8b26164447ba9044f